tests/unit/changeset-test.js

```diff
@@ -1057,6 +1057,60 @@ module('Unit | Utility | changeset', function(hooks) {
     assert.deepEqual(get(dummyModel, 'org'), expectedResult.org, 'should set value');
   });
 
+
+  /**
+   * #pendingChanges
+   */
+
+  test('#pendingChanges up to date if changeset is invalid', async function(assert) {
+    let dummyChangeset = Changeset(dummyModel, dummyValidator);
+    dummyChangeset.set('name', 'a');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), false, 'can be invalid');
+
+    let pendingChanges = dummyChangeset.pendingData;
+
+    assert.equal(get(pendingChanges, 'name'), 'a', 'should apply changes');
+  });
+
+  test('#pendingChanges up to date if changeset is valid', async function(assert) {
+    let dummyChangeset = Changeset(dummyModel);
+    dummyChangeset.set('name', 'foo');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), true, 'can be valid');
+
+    let pendingChanges = dummyChangeset.pendingData;
+
+    assert.equal(get(pendingChanges, 'name'), 'foo', 'should apply changes');
+  });
+
+  test('#pendingChanges up to date after saved changeset modified again', async function(assert) {
+    let dummyChangeset = Changeset(dummyModel);
+    dummyChangeset.set('name', 'foo');
+
+    assert.equal(get(dummyModel, 'name'), undefined, 'precondition');
+    assert.equal(get(dummyChangeset, 'isValid'), true, 'can be valid');
+
+    assert.equal(get(dummyChangeset.pendingData, 'name'), 'foo', 'should apply changes');
+    assert.equal(get(dummyModel, 'name'), undefined, 'original data is not applied');
+
+    dummyChangeset.save();
+
+    assert.equal(get(dummyModel, 'name'), 'foo', 'original data is updated');
+    assert.deepEqual(
+      dummyModel.getProperties('name', 'exampleArray'),
+      dummyChangeset.pendingData,
+      'pending changes equal to original data'
+    );
+
+    dummyChangeset.set('name', 'bar');
+
+    assert.equal(get(dummyChangeset.pendingData, 'name'), 'bar', 'pending data is returned in pendingChanges');
+    assert.equal(get(dummyModel, 'name'), 'foo', 'original data is not modified');
+  });
```

Comment:
Great job so far!  We probably need a test with an EmberData model - both pendingData on a belongsTo property and pendingData by swapping out a belongsTo relationship for another.  I'm skeptical the merge-deep util I made won't mutate the underlying model in certain cases.  Perhaps it doesn't; however, I don't have enough unit tests around `merge-deep` to be 100% confident.

Reply:
Good point, will add some more test cases

Reply:
(I like this change btw.  Great idea in place of the `force` option.